javalib/src/main/scala/scala/scalanative/nio/fs/UnixFileSystemProvider.scala

```diff
@@ -137,7 +138,8 @@ class UnixFileSystemProvider extends FileSystemProvider {
 
   private def getUserDir(): String = {
     val buff = stackalloc[CChar](4096)
-    val res  = unistd.getcwd(buff, 4095)
+    val res = unistd.getcwd(buff, 4095)
+    if (res == null) throw UnixException("Could not determine current working directory", errno.errno)
```

Comment:
Coding style nitpick: IMO it's better to always put statements and exceptions (with type `Unit` and `Nothing`) as separate lines in `if`s and `while`s. That makes them more visible.
```suggestion
    if (res == null)
      throw UnixException("Could not determine current working directory", errno.errno)
```